### PR TITLE
Dev/offsetlist compression

### PR DESCRIPTION
**Pull request by teisvh**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -52,4 +52,4 @@
 Cobham_hour.csv
 smartsendercpp.code-workspace
 /cobham_hour_reloaded.csv
-/cobham_hour_reloaded_new.csv
+/cobham_hour_reloaded_new.csv
```
Comment (karljohan12) [inline .gitignore]: Remove all .csv when *.csv

File: src/managers/TimestampManager.cpp [hunk 2]
```diff
@@ -2,11 +2,13 @@
 #include "TimestampManager.h"
 #include <iostream>
 
+
```
Comment (karljohan12) [inline src/managers/TimestampManager.cpp:5]: why space here

File: src/managers/TimestampManager.cpp [hunk 3]
```diff
@@ -172,4 +174,155 @@ TimestampManager::getTimestampsByGlobalId(int globID, int timestampA,
     res.push_back(allTimestampsReconstructed.at(count)); // Add last time stamp
 
     return res;
+}
+
+
```
Comment (karljohan12) [inline src/managers/TimestampManager.cpp:174]: Space

File: src/managers/TimestampManager.cpp [hunk 4]
```diff
@@ -172,4 +174,155 @@ TimestampManager::getTimestampsByGlobalId(int globID, int timestampA,
...
+        builder.currentByte = 0;
+        builder.remainingBits = 8;
+        builder.bytesCounter = 0;
+        const auto scheme = compressionSchemes.at(schemeID);
+
+        // Compress scheme ID.
+        // Always spend 8 bits on this value, as, when decompressing, we
+        // don't have a compression scheme for the first value.
+        // This allows for up to 256 different schemas.
+        appendBits(&builder, schemeID, 8);
```
Comment (karljohan12) [inline src/managers/TimestampManager.cpp]: schema!

File: src/managers/TimestampManager.cpp [hunk 5]
```diff
@@ -172,4 +174,155 @@ TimestampManager::getTimestampsByGlobalId(int globID, int timestampA,
...
+            for (const auto &t: temp) {
+                scheme(&builder, t);
+            }
+        }
+
+        // Update the chosen compression if it's better
+        size = builder.bytes.size();
+        if (size < bestCompression.size() || firstScheme) {
+            bestCompression = std::move(builder.bytes);
+            firstScheme = false;
```
Comment (karljohan12) [inline src/managers/TimestampManager.cpp]: Check for bestcompression empty, instead of first scheme?

File: src/managers/TimestampManager.cpp [hunk 6]
```diff
@@ -172,4 +174,155 @@ TimestampManager::getTimestampsByGlobalId(int globID, int timestampA,
...
+        }
+    }
+    return bestCompression;
+}
+
+std::vector<uint8_t>  TimestampManager::binaryCompressLocOffsets(std::unordered_map<int,
+        std::vector<std::pair<int, int>>>  offsets){
```
Comment (karljohan12) [inline src/managers/TimestampManager.cpp]: to much space before "offsets"

File: src/managers/TimestampManager.h [hunk 7]
```diff
@@ -53,9 +57,20 @@ class TimestampManager {
...
+
+    std::vector<uint8_t> binaryCompressLocOffsets(std::unordered_map<int,
+            std::vector<std::pair<int, int>>>  offsets);
 private:
     int timestampPrevious;
     bool readyForOffset = false;
     std::vector<int> allTimestampsReconstructed;
     std::vector<TwoLatestTimestamps> latestTimestamps;
+
+
```
Comment (karljohan12) [inline src/managers/TimestampManager.h]: nono

File: src/managers/TimestampManager.h [hunk 8]
```diff
@@ -53,9 +57,20 @@ class TimestampManager {
 
     std::vector<int> reconstructNTimestamps(int n);
 
+    void makeCompressionSchemes();
+
+    std::vector<uint8_t> binaryCompressGlobOffsets(const std::vector<std::pair<int,
+            int>>&
+    offsetsl);
+
+
```
Comment (karljohan12) [inline src/managers/TimestampManager.h]: why 2?

File: src/managers/TimestampManager.cpp (hunk 2 above)
Comment (teisvh) [reply src/managers/TimestampManager.cpp:5]: fixed

File: src/managers/TimestampManager.cpp (hunk 3 above)
Comment (teisvh) [reply src/managers/TimestampManager.cpp:174]: fixed

File: src/managers/TimestampManager.cpp (hunk 4 above)
Comment (teisvh) [reply src/managers/TimestampManager.cpp]: fixed

File: src/managers/TimestampManager.cpp (hunk 6 above)
Comment (teisvh) [reply src/managers/TimestampManager.cpp]: fixed

File: src/managers/TimestampManager.h (hunk 8 above)
Comment (teisvh) [reply src/managers/TimestampManager.h]: fixed

File: src/managers/TimestampManager.h (hunk 7 above)
Comment (teisvh) [reply src/managers/TimestampManager.h]: fixed

File: src/managers/TimestampManager.cpp (hunk 5 above)
Comment (teisvh) [reply src/managers/TimestampManager.cpp]: done

**Review by karljohan12 — APPROVED** [review]

**Review by karljohan12 — APPROVED** [review]

**Review by agneborn98 — APPROVED** [review]

**Pull request merged**